File: scala/src/main/scala/com/test/RestDB.scala
package com.test

import scala.util.Random

case class Texto(val id:Int, val texto:String)

class RestDB {

  private[this] var db = Map[Int,Texto]()
  
  def all = db.values
  
  def get(id:Int) = db.get(id)
  
  def create(text:String) = {
    val t = new Texto(math.abs(new Random() nextInt), text)
    
    db += (t.id -> t)
    
    t
  } 
  
  def set(existent:Texto, text:String) = {
    val t = new Texto(existent.id, text)
    
    db += (t.id -> t)
    
    t
  }
  
}

File: java/src/main/java/com/test/hellorest/RestDB.java
package com.test.hellorest;

import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import java.util.Random;

import com.test.hellorest.pojos.Texto;

public class RestDB {

	private Map<Integer, Texto> db = new HashMap<>();
	
	public Texto get(int id) {
		return db.get(id);
	}

	public Texto add(String texto) {
		if(texto != null) {
			Texto txt = new Texto(Math.abs(new Random().nextInt()), texto);
			db.put(txt.getId(), txt);
			
			return txt;
		}
		return null;
	}

	public Texto set(Texto txt) {
		
		if(txt.getId() == null || !db.containsKey(txt.getId())) {
			return null;
		}
		
		db.put(txt.getId(), txt);
		
		return txt;
	}

	public Collection<Texto> getAll() {
		return db.values();
	}

}
